### PR TITLE
Split MerkleTag into MerkleDigestTag and MerkleSigantureTag

### PR DESCRIPTION
**Pull request by m-renaud**
# Description of change

Split the `MerkleTag` type into separate `MerkleDigestTag` and `MerkleSignatureTag` types.

Previously, both signature and digest algorithms were represented with the same tag which didn't provide as much type safety as possible since they could be used interchangeably. Specifically, the `MerkleKey::extract_tags()` function previously returned `(MerkleTag, MerkleTag)` and one would need to look at the docs or implementation to determine which one was the signature and which was the digest which is error prone. Now these have separate types.

This change also revealed a (small) bug in the verifier where the digest tag was returned as part of a signature tag verification error which would have been very confusing.

## Links to any relevant issues

fixes issue #159 

## Type of change

- [x] Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

`cargo test`

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: identity-core/src/crypto/merkle_key/verifier.rs [hunk 1]
```diff
@@ -127,16 +128,16 @@ where
   D: MerkleDigest,
   S: MerkleSignature,
 {
-  let (tag_s, tag_d): (MerkleTag, MerkleTag) = MerkleKey::extract_tags(&key.merkle_key)?;
+  let (tag_s, tag_d): (MerkleSignatureTag, MerkleDigestTag) = MerkleKey::extract_tags(&key.merkle_key)?;
 
   // Validate the signature algorithm tag
   if tag_s != S::TAG {
-    return Err(Error::InvalidMerkleKeyTag(Some(tag_d)));
+    return Err(Error::InvalidMerkleSignatureKeyTag(Some(tag_s)));
```
Comment (l1h3r) [inline identity-core/src/crypto/merkle_key/verifier.rs:135]: 💯

**Review by l1h3r — CHANGES_REQUESTED** [review]

File: bindings/wasm/src/document.rs [hunk 2]
```diff
@@ -3,8 +3,9 @@
 
 use identity::core::decode_b58;
 use identity::core::FromJson;
+use identity::crypto::merkle_key::MerkleDigestTag;
 use identity::crypto::merkle_key::MerkleKey;
-use identity::crypto::merkle_key::MerkleTag;
+use identity::crypto::merkle_key::MerkleSignatureTag;
```
[the remark this answers is not in the document]
Comment (m-renaud) [reply bindings/wasm/src/document.rs]: Done.

File: identity-core/src/crypto/merkle_key/verifier.rs (hunk 1 above)
Comment (m-renaud) [reply identity-core/src/crypto/merkle_key/verifier.rs:135]: 😎

**Review by l1h3r — APPROVED** [review]
🚀

**Pull request merged**